### DISCUSSION TITLE
How To Make A Bug Report?

### DISCUSSION THREAD

**Discussion by TheMakerTales** [hlorus/CAD_Sketcher#107] [Announcements]
YAY! We are actually happy that you may have found a bug!
With your report, we might be able to make others' experiences even better.

Please follow the instructions below to make your report as valuable as possible to us.
The more you can do and fill in the more likely it is that we can solve the bug.

### 1. Is It A Bug?
Sometimes it can be tricky to distinguish a bug from a user error. One of the key ways to find out if it really is a bug is by asking others to recreate the issue. The best way to do this is over on the [Discord](https://discord.gg/EyRjypztmf). If it seems to be repeatable across multiple users of different knowledge levels then it probably is a bug.

Now by no means does it fully confirm that it's not a bug if others don't encounter it. It might be OS or even system based. If this is the case, the key thing here is to make sure that you can recreate the bug reliably on your system.

Lastly, please make sure that this is not a duplicate bug report by doing a quick little search in [issues](https://github.com/hlorus/CAD_Sketcher/issues). _Please note that duplicate bugs will be deleted._
<img width="627" alt="the search" src="https://user-images.githubusercontent.com/103122109/167103797-67d188fc-c8bd-4894-9405-511582b65a07.PNG">
Copy and paste in search: is:issue label:bug _Replace this text_

If you have any doubts feel free to bring it up in the [Discord](https://discord.gg/EyRjypztmf) first.

### 2. Reporting A Bug
Now that you know you have a bug head on over to [issues](https://github.com/hlorus/CAD_Sketcher/issues) and click "New issue".
<img width="133" alt="new issue" src="https://user-images.githubusercontent.com/103122109/167104495-35b20958-349a-4445-881c-0860c9c6ee8c.PNG">
You will need a GitHub account to be able to submit an issue.

If you would prefer not to make an account state it in the [Discord](https://discord.gg/EyRjypztmf) and someone else might be able to submit the issue on your behalf.
From here please select the "Bug report" template and fill it in, to the best of your ability. 

This is what the report looks like:
<img width="746" alt="Bug report" src="https://user-images.githubusercontent.com/103122109/167105647-9d72197a-9979-4955-ac59-88c5822a7a4a.PNG">

Please note while filling in the description you are able to drag and drop screenshots into the text field. Unfortunately, Github does not support drag and dropping of .blend files so if you would like to send us a file please use: [pasteall.org/blend/](https://pasteall.org/blend/)
<img width="517" alt="adding files" src="https://user-images.githubusercontent.com/103122109/167106588-5c3199db-5494-4864-bfbc-a82f533d4799.PNG">

### 3. Let us know all about it
Once you have posted your bug let the [Discord](https://discord.gg/EyRjypztmf) community know about it. Usually, the best place is in the bugs discussion channel.

At some point, a developer will get to your bug, they will usually reply via the Github issue. They might request some more information from you. e.g. What is your "System Console" saying. 

This can be found here:
<img width="450" alt="system console" src="https://user-images.githubusercontent.com/103122109/167108740-d9f9ac39-9631-48d5-b9cb-4dd3f8075c3c.PNG">

**Comment by JMGD-50** [comment]
Hello, I installed CAD Sketcher add-on in Blender 3.4.1. on my iMac M1 (Apple silicon). I see that it is installed and I also enabled it but I do not see in the tools bar on the right side of the Blender screen, I already tried to restart the computer but did not help.
What to do, or is it not yet ready for M1's
